### DISCUSSION TITLE
Flickering SK6812's in random areas [Raspberry PI + Dig Uno as level shifter setup]

### DISCUSSION THREAD

**Discussion by stickmanbrad** [awawa-dev/HyperHDR#488] [General]
Hi there,

Hoping you can help. I've been experiencing some extremely frustrating issues surrounding flickering led's in random patchy areas when watching any type of content. I have a bit of a different setup which I mimicked from another project. 

Setup specification:

- 217 Leds (Cool White SK6812's) around all 4 corners of my television
- Raspberry Pi 4 8GB (GPIO 18 passing to dig uno - used for level shifting and fuse protection, ground pin connected to ground wire from LED's)
- Meanwell 5V, 18A open power supply - Have tried powering both raspberry pi and led's, but since separated them with shared ground wire with pi having it's own power supply)
- The run from the Raspberry Pi to the Dig Uno is very short less than 5cm, but from the Dig Uno to the Led's is about 2m for the output positive, negative and data wire
- Zasluke Capture Card 4K 60 passthrough

Overall in most situations I have found the effect to be very pleasing. However, I do often get this flickering effect in random areas and it can be very distracting. I have tested all connections round the TV, of which appear to have good continuity. I have insured that the Dig Uno is set to 33Omhs (as a level shifter). It's very hard to figure out what the problem is as I can often see an LED just flickering. That appears to be the norm. But that moves to different led's. I have even replaced sections of LED's thinking maybe they were faulty.

Attached is the performance data and other pages from the dashboard. I did wonder what the high number in red next to the bin icon meant.

I would be really grateful for any insight into what might be causing this issue. I did even wonder whether I just need to move the PI, dig uno (used as level shifter) and capture card up by the tv for a shorter direct run to the LED's. Maybe that would help? But I was under the impression you can carry out a fairly long run up to 10m. I am using 12AWG which ensures there isn't much voltage drop.

I can provide logs if need be, but to tell you the truth there are not any errors that I can see.

Kind regards in advanced :)

![Capture1](https://user-images.githubusercontent.com/4136976/212477377-bec48344-05c1-427a-ae29-d43a292339a4.PNG)
![Capture2](https://user-images.githubusercontent.com/4136976/212477409-8ac8b90b-2c11-4a2e-aad9-b31138a40503.PNG)
![Capture3](https://user-images.githubusercontent.com/4136976/212477453-a09e840e-9a96-4fad-91eb-42c979d7a4ee.PNG)
![Capture4](https://user-images.githubusercontent.com/4136976/212477465-b1004569-a098-42eb-98ec-73fd95ac6204.PNG)

**Comment by stickmanbrad** [comment]
*Just a little update as I've been playing with some of the settings.

I went into the 'Image Processing' tab and came across the 'Anti-flickering threshold value'.

I then had a read and saw that it suggested a value of 32. I looked at the Led's behind my TV and found a Led that happened to be flickering. Applied the setting of 32 and it instantly went solid - No flicker. Now I'm not sure yet if that is the whole solution as some additional options then appeared ('Anti-flickering minimal step' & 'Anti-flickering timeout' - Will attach screenshot). Could this be related? - And why would it be required?

Although I do now get this in the logs: 'High resolution clock is NOT STEADY!'

Again, I'm not holding my breath that this is the solution and am still expecting flickering when I test more tonight. But thought I'd add this into the post.

![Capture5](https://user-images.githubusercontent.com/4136976/212478749-c8fbb20a-c396-403b-a0ca-10cd950279a7.PNG)

**Comment by awawa-dev** [comment]
Hi
2m data (voltage drop & EMI noise) and power cable (voltage drop) is definitely not a good thing. I would move at least the LED power supply and ESP32(HyperSerialESP32) closer to the LED strip. You can still have your Rpi 2m away from the ESP32, because such distance is perfectly safe for USB using good cable. Of course you still need a common ground between ESP and the LED strip. Unfortunately, the latest version of QuinLED-Dig-Uno has such an unfortunate construction and the fuse blocks the USB socket, so this ESP version cannot be used here.

Try to enable 'continues output' If it still works then anti-flickering filter is the solution and the problem in your case is the video source. Read more about this here: https://www.hyperhdr.eu/2021/04/how-to-set-up-hyperhdr-part-i-basic.html#chapterFlickering

**Comment by stickmanbrad** [comment]
> Hi 2m data (voltage drop & EMI noise) and power cable (voltage drop) is definitely not a good thing. I would move at least the LED power supply and ESP32(HyperSerialESP32) closer to the LED strip. You can still have your Rpi 2m away from the ESP32, because such distance is perfectly safe for USB using good cable. Of course you still need a common ground between ESP and the LED strip. Unfortunately, the latest version of QuinLED-Dig-Uno has such an unfortunate construction and the fuse blocks the USB socket, so this ESP version cannot be used here.
> 
> Try to enable 'continues output' If it still works then anti-flickering filter is the solution and the problem in your case is the video source. Read more about this here: https://www.hyperhdr.eu/2021/04/how-to-set-up-hyperhdr-part-i-basic.html#chapterFlickering

Hiya,

Thank you so much for the quick reply. Yes - I had a feeling that what I've done might not be ideal.. And that option is still on the table at last resort. In regards to the 'continues output', would I still leave those anti-flickering settings as mentioned in previous screenshot? Or disable that, and enable 'continues output'?

Furthermore, the part where you mentioned about being able to have the raspberry Pi apart from the dig uno, power supply and Led's, wouldn't be possible because I'd still need gpio 18 (data) and the ground as you mentioned nearby? There are no USB's involved other than for the capture card. I have taken the ESP32 off of the dig uno so that the raspberry pi is purely running things through the built in level shifter and fuse protection. That's the only reason i'm using the dig uno.

In regards to those performance figures, are you able to also let me know what the red number next to the bin means? - I've also noticed the data numbers that get pushed don't seem to be coming back with the same amount. So does it seem likely that the long run is having an effect on that? - Because i've seen some users have say 3000 go to the Led's and then 3000 come back.

Again, huge thanks in advanced.
Brad

**Reply by idarius** [reply]
hello @stickmanbrad,

Sorry to dig up an old post but i've some questions for you if you don't mind.

- Have you still your flickering issue with this setup ? 
- Can you post a picture of your wiring between the Rpi, esp, dig uno and your power supply ?

I have a dig uno that i want to use for the fuse and level shifter, but i don't find any exemple.

Thank you !

**Comment by awawa-dev** [comment]
> Thank you so much for the quick reply. Yes - I had a feeling that what I've done might not be ideal.. And that option is still on the table at last resort. In regards to the 'continues output', would I still leave those anti-flickering settings as mentioned in previous screenshot? Or disable that, and enable 'continues output'?

Antiflickering filter may reduce refresh rate (because it eliminates noise, minimal changes on dark park are not updated on the led strip,) so to make sure I asked you to enable  'continues output' that forces maximum refresh rate on the LED strip to confirm that antiflickering is the real solution. You should disable 'continues output' (it sends a frame even if it is not changed) when you finish testing.

> wouldn't be possible because I'd still need gpio 18 (data) and the ground as you mentioned nearby? 

in my proposed solution you should use Adalight driver in HyperHDR which communicates with the ESP32 (HyperSerialESP32 firmware should be uploaded on ESP32) over USB. The common ground is between ESP32 and the LED strip ground in such case (it will be forwarded over USB to Rpi also...but it is not important).

> In regards to those performance figures, are you able to also let me know what the red number next to the bin means? - I've also noticed the data numbers that get pushed don't seem to be coming back with the same amount. So does it seem likely that the long run is having an effect on that? - Because i've seen some users have say 3000 go to the Led's and then 3000 come back.

Not sure which HyperHDR version from which date you are using: there was a problem with statistics when the LED device 'Refresh time' was set. But it should be fixed in the latest. Regardless of the statistics problem: 'Refresh time' should be set to 0 for better performance of ws281x or adalight awa driver because these are high performance drivers. It generates unnecessary delay only.

**Comment by awawa-dev** [comment]
> Hiya,
What I did was I re flashed the esp32 which comes with the dig uno with that firmware. The led output is typically on gpio 16 (hence why I changed the config. I then ran that into the led pin in the dig uno. I also put a ground lead from that esp32 to the LEDs. It did work but still had to put the anti flickering settings on it..
Do you agree that it’s probably best I still move everything closer to the tv with shorter runs?
Kind regards

To keep it in one place I move it to the HyperHDR discussion.
It may help other users also.

To clarify: your problem was caused by the your video source because the anti-flickering filter solves it.
It was not caused by high refresh rate and/or EMI interference of data signal on long cable, because even if you enabled the 'continues output' it still worked for you.

If you are happy with the current result you can leave it as it is. It's good that you dropped that GPIO18 and direct driving ws218x from Rpi and you are using Adalight now, because generating signal on Rpi is not a cheap thing: it takes a lot of real time resources which other drivers or applications (including HyperHDR) need. Still 2 meters data & power cables is not a good thing.

**Reply by awawa-dev** [reply]
BTW probably you want to enable anti-flickering timeout also...e.g. 300 or 500. If you set it too low the flickering may come back.

**Reply by stickmanbrad** [reply]
> BTW probably you want to enable anti-flickering timeout also...e.g. 300 or 500. If you set it too low the flickering may come back.

Hey again,

Just wanted to add a few things.

So in regards to the anti-flickering timeout, I have now set this to 500ms. 

I still unfortunately am getting flickering, but it appears to take place in slightly dimmer scenes. I thought it had stopped, but I think it was just the individual led flicker I saw here and there. Green is a common colour which I will see flickering. Just wondered if there were any other settings which might effect this?

However, as a last ditch attempt I’m going to try moving everything closer to the LED’s tomorrow, so should have an update on this as per your guidance.

In regards to the included ESP32, I think you were correct about the Dig Uno model having a lower transfer rate of data. Mainly because I’m seeing the following message:

‘Could not detect HyperSerialEsp8266/HyperSerialESP32 device’

I do have another ESP32 & ESP8266 which I purchased for other projects so wonder if that might work better?

So it’s where I go from here, try shortening the leads or expect some flickering (if that is normal sometimes?). I am using the latest beta version of the software.

Kind regards in advanced,

**Reply by awawa-dev** [reply]
Before changing anything try to increase 'anti-flickering threshold value' from 32 to 255 (maximum) and test it again. RGB(32,32,32) is quite dark gray so if the flickering appears on slightly brighter scene it may happen because the filter is disabled for brighter colors.
I think you can still use Dig Uno with HyperSerialESP32: just compile it online with 921600 speed and set that speed also in HyperHDR.

**Reply by awawa-dev** [reply]
'anti-flickering minimal step'  may be also set e.g. to 6 as the last resort. just to confirm the video source.

**Reply by awawa-dev** [reply]
Some users reported flickering when use ws281x driver to drive LED strip from Rpi (one more argument why this driver is not officially supported) but it happened for longer LEDs strip than yours > ~250LEDs. And this driver has a tendency to stop working at all over ~300LEDs. It's hard to maintain strict Neopixel timings for the real time system like Rpi.

**Reply by stickmanbrad** [reply]
> 'anti-flickering minimal step' may be also set e.g. to 6 as the last resort. just to confirm the video source.

Heya,

Thanks again for the reply. So I tried the ‘anti-flickering minimal step’ option, and changed it from 2 to 6. I immediately saw more flickering. I’m assuming that was expected to show it’s the video source problem? Is that the case? 

If so, one thing I wanted to ask you was is it down to this zasluke 4K capture card? - and if this is the case, should I consider a new capture card. I noticed you recently reviewed the elgato model. Do you know at all whether it supports hdcp 2.2 as I cannot seem to confirm this.

I also am still unfortunately getting some flashing green light when it’s a dark scene. Not sure if this related to the capture card again?

look forward to your reply

Kind Regards as always.

**Reply by awawa-dev** [reply]
Not sure. For hdcp 2.2 you will need a HDMI splitter that supports it.

**Reply by stickmanbrad** [reply]
> Not sure. For hdcp 2.2 you will need a HDMI splitter that supports it.

With that setting being set to 6 was there supposed to be more flickering if it was a video capture issue? I’m not sure if that was a test or not?

In regards to the Elgato, after your review it did seem like it was very impressive. Were you happy with the results, and would you recommend as an alternative? - would this help resolve the green in low light?


Kind regards

**Reply by awawa-dev** [reply]
I'm not sure because I don't see it live an I don't know what it cause it or it's a subjective impression. That setting should reduce the effect. Video source is HDR or not? Did you calibrated your grabber for HDR and use dedicated LUT?

**Reply by awawa-dev** [reply]
And first: ws218x driver from Rpi directly. I really should stop here.

**Reply by stickmanbrad** [reply]
> And first: ws218x driver from Rpi directly. I really should stop here.

Heya, so first thing I should mention is since your recommendation I moved over to the adafruit ESP32 option so that is out of the question now.

In regards to the above I’m definitely still seeing some flashing. Majority of this is in lower lit areas. Green will then appear when they should probably just be off. The video source itself is set the HDR and the content is HDR. As far as I know, I’m using the pre-installed LUT and not a custom generated one. Unfortunately there isn’t one for this particular capture card. However, I am aware that there is a method to generate a LUT table, so this is something I would be looking to do this afternoon. Is this something that would help with these darker colour issues?

I was reading your blog, and noticed that it suggested maybe changing the Backlight threshold from 1% to maybe 2/3. Could this also help with the weird green flashing in darkness?

Again I appreciate your time and expertise in helping diagnose. I know it’s slightly harder not being able to physically see the issue :P.

Brad

**Reply by awawa-dev** [reply]
Do you have Backlight threshold enabled? Does the LED strip is always enabled or sometimes it goes completely off on dark scenes?

Test it and set RGB(0,0,0) in the Remote control: Effects (color).

**Reply by stickmanbrad** [reply]
> Do you have Backlight threshold enabled? Does the LED strip is always enabled or sometimes it goes completely off on dark scenes?
> 
> Test it and set RGB(0,0,0) in the Remote control: Effects (color).

Hiya, Apologies for the delay.

In answer to your question - Yes, it is currently set to 1%

I find that the LED’s will sometimes appear off on dark scenes, but other times on dark scenes there will be some flickering of green on a small area. I have now ensured that the LUT calibration has taken place, and I believe it did manage to take the calibration through my Windows 10 laptop connected to the capture source (UHD enabled on laptop) as per instructions. I cannot say I have noticed much of a difference. But the flickering is still fairly visible to the point I had to turn it off last night.

One thing I did want to run by you (which may help others), obviously you requested I set the ‘Refresh Time’ to 0. With this set it is noticeably delayed on the TV. What I mean by this is that the LED’s actually not react quicker than the TV image does. This means if a hand moves by for example, the TV is delayed. This got me thinking about the capture card’s pass through which obviously is currently connected to the TV. I read some more of your Q&A and discussion and wondered whether I do need to put a splitter in place before hand. I wondered whether this might make things match latency wise?

I was looking at the following which as 1 input (say the chrome cast), and then two outputs (1 scaling down to 1080P for the capture card, and 2 for 4k pass back to to). Because currently, I’m passing 4K 60 into the capture card and then obviously scaling it back in HyperHDR to 720p. But the passthrough is still 4K. Thinking maybe there’s latency. Could I get your input on this please?

https://www.amazon.co.uk/HDMI-Splitter-1x2-4K-60Hz/dp/B083JVPXDY/ref=sr_1_1_sspa?keywords=ezcoo+hdmi+splitter&qid=1674031921&sprefix=Ezco%2Caps%2C132&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1

Although I’m still not sure quite what to do about the flickering and green in dark scenes. 

Kind Regards

**Reply by stickmanbrad** [reply]
In regards to the Black effects test. After putting RGB 0,0,0 in, there is no LED’s on that I can see. Total darkness.

Not sure what that might indicate.

**Reply by awawa-dev** [reply]
It's important. Try to increase Backlight threshold to 2 or 3 and test it again. Probably 1 is too low for your setup and real black color RGB(0,0,0) disables anti-flickering filter.

**Reply by stickmanbrad** [reply]
> It's important. Try to increase Backlight threshold to 2 or 3 and test it again. Probably 1 is too low for your setup and real black color RGB(0,0,0) disables anti-flickering filter.

Hiya,

So increased to ‘2’, and did the RGB(0,0,0) test and still no lights. Then went to ‘3’, which then provided a low white glow. Is that what we are looking for?

Apologies I’m not quite understanding what this means.

Kind Regards

**Reply by awawa-dev** [reply]
Yes, stay with Backlight threshold at 3 and re-test anti-flickering filter. Previously if some RGB value from the video source was fluctuating around RGB(0,0,0) the anti-flickering filter could not prevent it from changing rapidly.

**Reply by stickmanbrad** [reply]
> Yes, stay with Backlight threshold at 3 and re-test anti-flickering filter. Previously if some RGB value from the video source was fluctuating around RGB(0,0,0) the anti-flickering filter could not prevent it from changing rapidly.

Thank you, I’ll test that and see how we get on. I use the HA hyperhdr app to turn down the brightness of the LED’s as I usually find them too bright. I do this pretty much every time I boot up. Will the lower brightness change what is needed here for the backlight threshold. The reason I say this is I notice it tends to default to much brighter quite frequently (like a brightness reset or something).

Also in regards to the latency, I mentioned that splitter above. Would that be something I should be looking at, or is the loop-out going to behave just the same? 

Initial indications suggest that the ‘Backlight threshold’ has helped with the latency problems, but might not be related at all.

Kind regards

**Reply by awawa-dev** [reply]
> One thing I did want to run by you (which may help others), obviously you requested I set the ‘Refresh Time’ to 0. 

Make sure your refresh rate in the smoothing is set to reasonable value. I really don't know what happens to that unsupported LED driver: like I said it's not a cheap thing to emulate Neopixel signal on Rpi. There is no such problems with highspeed AWA Adalight or HyperSPI which work without setting additional delays. I consider at least this subject as closed and I won't come back to it.

**Reply by stickmanbrad** [reply]
> > One thing I did want to run by you (which may help others), obviously you requested I set the ‘Refresh Time’ to 0.
> 
> Make sure your refresh rate in the smoothing is set to reasonable value. I really don't know what happens to that unsupported LED driver: like I said it's not a cheap thing to emulate Neopixel signal on Rpi. There is no such problems with highspeed AWA Adalight or HyperSPI which work without setting additional delays. I consider at least this subject as closed and I won't come back to it.

Hiya,

Apologies for posting on this again. I was still continuing to have issues with the ESP32. I managed to find a different ESP32. In fact I had actually managed to find a CH9102x, and I noticed this could work over 4MB speed. So I changed the config for that speed and off to the races I went.

Plugged it all in, and it works but I'm getting that dreadful 'Could not detect HyperSerialEsp8266/HyperSerialESP32 device' in my logs. But it does seem to be working on the 4MB speed. I'll attach logs below and if this needs moving I can do so. I wondered whether maybe that error message is specifically related to go off if it's not 2MB? Could It be at all possible it is working? Even running the 'stty -F' command on the port says 4MB.

Thanks again

_Logs Beginning_

2023-01-19T19:23:35.913Z [LEDDEVICE_ADALIGHT] Opening UART: ttyACM0
2023-01-19T19:23:35.913Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:183) _rs232Port.open(QIODevice::ReadWrite): ttyACM0, Baud rate [4000000]bps
2023-01-19T19:23:35.957Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:188) portName:          ttyACM0
2023-01-19T19:23:35.957Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:189) systemLocation:    /dev/ttyACM0
2023-01-19T19:23:35.958Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:190) description:       USB Single Serial
2023-01-19T19:23:35.958Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:191) manufacturer:      1a86
2023-01-19T19:23:35.958Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:192) productIdentifier: 0x55d4
2023-01-19T19:23:35.958Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:193) vendorIdentifier:  0x1a86
2023-01-19T19:23:35.958Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:194) serialNumber:      5377014907
2023-01-19T19:23:36.214Z [LEDDEVICE_ADALIGHT] Could not detect HyperSerialEsp8266/HyperSerialESP32 device
2023-01-19T19:23:36.215Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:329) Switch on
2023-01-19T19:23:36.215Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:405) Power On
2023-01-19T19:23:36.215Z [COMPONENTREG0] LED device: enabled
2023-01-19T19:23:38.045Z [WEBSOCKET] (JsonAPI.cpp:1250) log streaming activated for client ::ffff:192.168.1.84
2023-01-19T19:24:29.938Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 8, hits: 3582, needed: 1, cleanup: 0, limit: 8
2023-01-19T19:25:29.930Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 8, hits: 3596, needed: 0, cleanup: 0, limit: 8
2023-01-19T19:25:29.936Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3596], [LED0: FPS = 50.00, send = 3000, processed = 3000, dropped = 0]
2023-01-19T19:26:29.927Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 7, hits: 3596, needed: 0, cleanup: 1, limit: 8
2023-01-19T19:26:29.941Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3597], [LED0: FPS = 50.00, send = 3000, processed = 3000, dropped = 0]
2023-01-19T19:27:19.227Z [SMOOTHING0] Clearing queued colors before: disabling
2023-01-19T19:27:19.227Z [SMOOTHING0] Smoothing queue is cleared
2023-01-19T19:27:19.228Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:138) Disable device
2023-01-19T19:27:19.229Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:355) Switch off
2023-01-19T19:27:19.229Z [LEDDEVICE_ADALIGHT] (LedDevice.cpp:313) Set LED strip to black/power off
2023-01-19T19:27:19.232Z [WEBSOCKET] (JsonAPI.cpp:1259) log streaming deactivated for client  ::ffff:192.168.1.84
2023-01-19T19:27:19.232Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:123) Flush was successful
2023-01-19T19:27:19.234Z [COMPONENTREG0] LED device: disabled
2023-01-19T19:27:19.235Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3597]
2023-01-19T19:27:19.240Z [LEDDEVICE_ADALIGHT] Received: '
HyperHDR frames: 50 (FPS), receiv.: 49, good: 49, incompl.: 0, mem: 3268, heap: 315380
RGBW => Gain: 255/255, red: 128 , green: 128 , blue: 128' (146)
2023-01-19T19:27:19.241Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:106) Close UART: ttyACM0
2023-01-19T19:27:20.376Z [WEBSOCKET] (JsonAPI.cpp:1250) log streaming activated for client ::ffff:192.168.1.84
2023-01-19T19:27:29.935Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3597, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3596]
2023-01-19T19:27:29.936Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 7, hits: 3597, needed: 0, cleanup: 0, limit: 8
2023-01-19T19:28:29.932Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3596]
2023-01-19T19:28:29.932Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 6, hits: 3596, needed: 0, cleanup: 1, limit: 8
2023-01-19T19:29:29.924Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 6, hits: 3596, needed: 0, cleanup: 0, limit: 8
2023-01-19T19:29:29.938Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 11ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3597]
2023-01-19T19:29:41.470Z [IMAGETOLED0] Total index number is: 240640 (memory: 240640). User sparse processing is: disabled, image size: 1280 x 720, area number: 217
2023-01-19T19:30:29.937Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 4, hits: 3597, needed: 0, cleanup: 1, limit: 8
2023-01-19T19:30:29.939Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 14ms, frames = 3597, invalid = 0], [INSTANCE0: FPS = 59.93, processed = 3596]
2023-01-19T19:31:29.930Z [V4L2:USB3.0 HD VIDE] Video cache: enabled, size: 4, hits: 3596, needed: 0, cleanup: 0, limit: 8
2023-01-19T19:31:29.930Z [PERFORMANCE] [USB capturing: FPS = 59.94, decoding = 14ms, frames = 3596, invalid = 0], [INSTANCE0: FPS = 59.94, processed = 3596]

_Logs ending_
